prow/clonerefs/run.go

```diff
@@ -73,3 +87,49 @@ func (o Options) Run() error {
 
 	return nil
 }
+
+// addSSHKeys will start the ssh-agent and add all the specified
+// keys, returning the ssh-agent environment variables for reuse
+func addSSHKeys(paths []string) ([]string, error) {
+	vars, err := exec.Command("ssh-agent").CombinedOutput()
+	if err != nil {
+		return []string{}, fmt.Errorf("failed to start ssh-agent: %v", err)
+	}
+	logrus.Info("Started SSH agent")
+	// ssh-agent will output three lines of text, in the form:
+	// SSH_AUTH_SOCK=xxx; export SSH_AUTH_SOCK;
+	// SSH_AGENT_PID=xxx; export SSH_AGENT_PID;
+	// echo Agent pid xxx;
+	// We need to parse out the environment variables from that.
+	parts := strings.Split(string(vars), ";")
+	env := []string{strings.TrimSpace(parts[0]), strings.TrimSpace(parts[2])}
+	for _, keyPath := range paths {
+		// we can be given literal paths to keys or paths to dirs
+		// that are mounted from a secret, so we need to check which
+		// we have
+		if err := filepath.Walk(keyPath, func(path string, info os.FileInfo, err error) error {
+			if info == nil {
+				return nil
+			}
+			if info.IsDir() {
+				if strings.HasPrefix(info.Name(), "..") {
```

Comment:
We need to skip files that start with ".." in addition to directories (return `nil` for files instead of `filepath.SkipDir`).